instrumentation/netty/netty-4.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/netty/v4_0/client/Netty40ClientSslTest.java

```diff
@@ -0,0 +1,244 @@
+  @Test
+  public void shouldSuccessfullyEstablishSslHandshake() throws Exception {
+    // given
+    Bootstrap bootstrap =
+        createBootstrap(eventLoopGroup, Arrays.asList("TLSv1", "TLSv1.1", "TLSv1.2", "TLSv1.3"));
+
+    URI uri = server.resolveHttpsAddress("/success");
+    DefaultFullHttpRequest request =
+        new DefaultFullHttpRequest(
+            HttpVersion.HTTP_1_1, HttpMethod.GET, uri.getPath(), Unpooled.EMPTY_BUFFER);
+    HttpHeaders.setHost(request, uri.getHost() + ":" + uri.getPort());
+
+    AtomicReference<Channel> channel = new AtomicReference<>();
+    // when
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          try {
+            channel.set(bootstrap.connect(uri.getHost(), uri.getPort()).sync().channel());
+            CompletableFuture<Integer> result = new CompletableFuture<>();
+            channel.get().pipeline().addLast(new ClientHandler(result));
+            channel.get().writeAndFlush(request).get(10, TimeUnit.SECONDS);
+            result.get(10, TimeUnit.SECONDS);
+          } catch (Exception ex) {
+            throw ex;
+          } finally {
+            if (channel.get() != null) {
+              channel.get().close();
+            }
+          }
+        });
+
+    // then
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent"),
+                span -> {
+                  span.hasName("CONNECT").hasKind(INTERNAL).hasParent(trace.getSpan(0));
+                  span.hasAttributesSatisfyingExactly(
+                      equalTo(SemanticAttributes.NETWORK_TRANSPORT, "tcp"),
+                      equalTo(SemanticAttributes.NETWORK_TYPE, "ipv4"),
+                      equalTo(SemanticAttributes.SERVER_ADDRESS, uri.getHost()),
+                      equalTo(SemanticAttributes.SERVER_PORT, uri.getPort()),
+                      equalTo(NetworkAttributes.NETWORK_PEER_PORT, uri.getPort()),
+                      equalTo(NetworkAttributes.NETWORK_PEER_ADDRESS, "127.0.0.1"));
+                },
+                span -> {
+                  span.hasName("SSL handshake").hasKind(INTERNAL).hasParent(trace.getSpan(0));
+                  span.hasAttributesSatisfyingExactly(
+                      equalTo(SemanticAttributes.NETWORK_TRANSPORT, "tcp"),
+                      equalTo(SemanticAttributes.NETWORK_TYPE, "ipv4"),
+                      equalTo(NetworkAttributes.NETWORK_PEER_PORT, uri.getPort()),
+                      equalTo(NetworkAttributes.NETWORK_PEER_ADDRESS, "127.0.0.1"));
+                },
+                span -> {
+                  span.hasName("GET").hasKind(CLIENT).hasParent(trace.getSpan(0));
+                },
+                span -> {
+                  span.hasName("test-http-server").hasKind(SERVER).hasParent(trace.getSpan(3));
+                }));
+    // cleanup
+    if (channel.get() != null) {
+      channel.get().close();
```

Comment:
groovy test uses `channel?.close()?.sync()`
channel is also close in a finally block further up, I guess closing it here isn't needed